From nobody Mon Sep 17 00:00:00 2001
From: Keith Busch <kbusch@meta.com>
Subject: [PATCHv2 2/2] blk-mq: allow cached requests to be used for flush operations

Rebased against the series tree: an earlier carried patch shifted the
surrounding context. The author's change lines are unmodified.

diff --git a/block/blk-mq.c b/block/blk-mq.c
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -660,8 +660,6 @@
 			return NULL;
 
 		if (blk_mq_get_hctx_type(opf) != rq->mq_hctx->type)
-			return NULL;
-		if (op_is_flush(rq->cmd_flags) != op_is_flush(opf))
 			return NULL;
 		if (blk_rq_is_passthrough(rq) != blk_op_is_passthrough(opf))
 			return NULL;
@@ -3077,8 +3075,6 @@
 	if (type != rq->mq_hctx->type &&
 	    (type != HCTX_TYPE_READ || rq->mq_hctx->type != HCTX_TYPE_DEFAULT))
 		return NULL;
-	if (op_is_flush(rq->cmd_flags) != op_is_flush(opf))
-		return NULL;
 	if (blk_rq_is_passthrough(rq) != blk_op_is_passthrough(opf))
 		return NULL;
 	rq_list_pop(&plug->cached_rqs);
